Storybook v8 stories batch 5

misama-ct opened the pull request:
## Background
In an effort to switch from Storybook v5 to Storybook v8, the existing stories had to be migrated to a new storybook-format and converted to typescript. 

This pull-request is [part of a batch](https://github.com/commercetools/ui-kit/pull/2875). It contains the typescript-equivalents of existing storybook v5 stories and aims to replicate the same functionality / value. 

## Goal
Feature parity between v8 and v5 storybook. After merging all batches, a product developer who looked at the storybook v5 version yesterday, should be able to look at the v8 version tomorrow and be as productive (or more productive) as before. 

## Review instructions
A thorough **code review** is not required yet. The code was either copied from the existing js-equivalent or quickly written from scratch to replicate what was already there to fit the new story-format. It is expected that those stories contain typescript errors, introduced by the conversion or previously existing. (Another pull-request will take care of those issues at a later stage).

What is expect is a **story review**:

You will need to compare the v5 with the v8 storybook (ideally side-by-side) and make sure that every component in this pull-request has:

- [ ] a readme-file in the same or a parent-folder
- [ ] a props-table displaying available props (and appropriate inputs for them)
- [ ] one or more stories that showcase the same or more functionality as the v5 equivalent

You will find the links to the different storybooks below. 


> If something is missing or irritating, add a comment to the source file here in the pull-request to start a conversation.

FilPob commented:
In the DateInput Story the calendar dropdown is cut. Otherwise looks all good. 
<img width="1138" alt="image" src="https://github.com/user-attachments/assets/d0794b9b-4c81-4960-ad6f-9d227001d34e">

ddouglasz commented:
Thanks! 🙌🏽
At the moment, we can only interact with the `CheckboxInput` using the `controls`(formerly `knobs`). Do you think we should at least allow the user to click on the checkbox in the basic example to toggle the state from checked to unchecked too?

misama-ct commented:
> Thanks! 🙌🏽 At the moment, we can only interact with the `CheckboxInput` using the `controls`(formerly `knobs`). Do you think we should at least allow the user to click on the checkbox in the basic example to toggle the state from checked to unchecked too?

I think so, it's certainly irritating if it does not toggle when clicked. I'll adjust the story or create an additional one.

ddouglasz commented:
**CreatableSelectInput**

isMulti is false by default in v5 unless the user switches it to true.

isValidNewOption is likely missing the expected value so it breaks the input too. 
![2024-08-13 13 16 27](https://github.com/user-attachments/assets/8f4e7f9b-1db7-4854-a5d4-e67ee4f96ff1)

ddouglasz commented:
**LocalizedMultilineTextInput**
could you please check the formatting of this again in the README? Looks like it is missing the javascript formatting, unlike the others.

<img width="1127" alt="image" src="https://github.com/user-attachments/assets/1578d5cb-6bbc-4882-96dd-04a58769c9f7">

ddouglasz commented:
LocalizedRichTextInput: Basic example
I think we should make the `selectedLanguage` control a select dropdown with already known options, an text input allows the user to put any value, and that seems to break the story.
![2024-08-13 18 27 53](https://github.com/user-attachments/assets/94a101a9-1272-48cb-bd14-ce5d2431c082)

ddouglasz commented:
LocalizedTextInput:
We also need to make the `selectedLanguage` a dropdown here too

ddouglasz commented:
**MultilineTextInput**

1. The `rightActionIcon` is meant to display when `hasRightAction` is set to true. Mainly for storybook usage. It seems like that is what the `rightActionProps` is doing now. But it also behaves like we need to set an object as its value when in reality, it acts as a boolean because just merely toggling it gives us the behaviour we want. 
Pardon me if I may be missing some context, otherwise maybe want to use the `hasRightAction` boolean as it was in SB v5.

V5 
![2024-08-13 18 57 54](https://github.com/user-attachments/assets/fe7952e4-d03e-491d-bc71-9cde340ebc68)

V8
![2024-08-13 18 59 07](https://github.com/user-attachments/assets/ded16238-a24d-4014-94b8-3402b83ed0f1)

misama-ct commented:
@FilPob I fixed the DateInput issue, please have another look

@ddouglasz 

**CheckboxInput** 
✅  Fixed, the checkbox can be toggled now (in the basic example).


**CreatableSelectInput**
✅ `isMulti` is set to false by default now
✅  `isValidNewOption` can not be set anymore (since it requires a function)

**LocalizedMultilineTextInput**
> could you please check the formatting of this again in the README? Looks like it is missing the javascript formatting, unlike > the others.

✅ Fixed. In the docs/readme file the code-block was annotated as `js` code while actually being `jsx`code 

**LocalizedRichTextInput + LocalizedTextInput**
> [...] we should make the selectedLanguage control a select dropdown [...]

✅ `selectedLanguage` is now a dropdown including the 3 available languages for this example

![image](https://github.com/user-attachments/assets/90461620-a38a-45d1-87d1-dbc6b05c043a)


**MultilineTextInput**
> 
> 1. The `rightActionIcon` is meant to display when `hasRightAction` is set to true. Mainly for storybook usage. It seems like that is what the `rightActionProps` is doing now. But it also behaves like we need to set an object as its value when in reality, it acts as a boolean because just merely toggling it gives us the behavior we want.
>    Pardon me if I may be missing some context, otherwise maybe want to use the `hasRightAction` boolean as it was in SB v5.

I didn't want to introduce the `hasRightAction` toggle cause so far the prop-tables of all components only show actual props that a component has to offer (and it should stay that way). 

I updated the TypeScript annotation for `rightActionIcon` so that it only appears if `rightActionProps` is also set.

![image](https://github.com/user-attachments/assets/a34da306-22df-42c8-8b5b-ceec6a4f6bbd)

ddouglasz commented:
> I updated the TypeScript annotation for rightActionIcon to clarify that it only shows up if rightActionProps is also set.
Yes, this alternative makes sense to me. Thank you.

ddouglasz reviewed (approved):
Thank you 🙌🏽

ByronDWall commented on packages/components/inputs/creatable-select-input/src/creatable-select-input.stories.tsx:
```diff
@@ -0,0 +1,164 @@
+  hasWarning: false,
+  backspaceRemovesValue: true,
+  isClearable: false,
+  isSearchable: true,
```
not in scope for this PR, but why even make `isSearchable` a prop if there's no way to create a new option if it's `false`?

ByronDWall commented on packages/components/inputs/date-time-input/src/date-time-input.readme.mdx:
```diff
@@ -0,0 +1,6 @@
+import { Meta, Markdown } from '@storybook/blocks';
```
The `props` page for `DateTimeInput` is missing most of the props
![Screenshot 2024-08-19 at 10 54 52 AM](https://github.com/user-attachments/assets/22760c97-0947-4756-ae10-02166ae06b24)

misama-ct replied:
This happened before, sb has issues parsing the types from class-based components. I [added a wrapper-component](https://github.com/commercetools/ui-kit/pull/2881/commits/e943fef6605e26e0738a9136e9c60105a759ae61) for storybook. Alternatively one could have manually added each prop-control via the `argTypes` config of the story, but then you'd have to remember keeping those in sync when updating the component.

ByronDWall commented on packages/components/inputs/localized-multiline-text-input/src/localized-multiline-text-input.stories.tsx:
```diff
@@ -0,0 +1,80 @@
+  const [value, setValue] = useState({
+    en: 'Horse\nCow\nDuck',
+    de: 'Pferd\nKuh\nEnte',
+    'nan-Hant-TW': '马\n奶牛\n鸭子',
```
nit, Hant-TW is traditional, these translations are in simplified
```suggestion
    'nan-Hant-TW': '馬\n奶牛\n鴨子',
```

ByronDWall reviewed (approved):
Overall looks good to me, once again definitely an improvement
